Migrate backends/arm to the new namespace

dbort opened the pull request:
Summary:
Move the Arm backend out of the `torch::` namespace, and update to avoid using the `torch::` or `exec_aten::` namespaces.

Also, move the VelaBinStream code into a namespace to avoid declaring symbols in the global namespace.

Differential Revision: D63923290

zingo commented:
PR seem to miss a fix here
[trunk / test-arm-backend-delegation / linux-job](https://hud.pytorch.org/pr/pytorch/executorch/5904#31113444440) ([gh](https://github.com/pytorch/executorch/actions/runs/11191018932/job/31113444440))
/pytorch/executorch/backends/arm/runtime/ArmBackendEthosU.cpp:161:19: error: 'is_contiguous_dim_order' was not declared in this scope

dbort committed:
Migrate backends/arm to the new namespace (#5904)

Summary:

Move the Arm backend out of the `torch::` namespace, and update to avoid using the `torch::` or `exec_aten::` namespaces.

Also, move the VelaBinStream code into a namespace to avoid declaring symbols in the global namespace.

Reviewed By: mergennachin

Differential Revision: D63923290

dbort commented:
@zingo Thanks; I just pushed a new version that fixes that issue, and I verified it locally.

dbort commented:
@pytorchbot cherry-pick --onto release/0.4 -c fixnewfeature

zingo commented:
This is already closed but If the CI was happy when adding trunk builds I'm also happy, thanks for fixing this before merging.
